modules/cudacodec/src/video_reader.cpp

```diff
@@ -399,6 +426,10 @@ namespace
     }
 
     bool VideoReaderImpl::get(const int propertyId, double& propertyVal) const {
+        if (propertyId == cv::VideoCaptureProperties::CAP_PROP_POS_FRAMES) {
+            propertyVal = static_cast<double>(iFrame);
+            return true;
+        }
```

Comment:
I have not found where `iFrame` is incremented besides the cese with initial seeking.  CAP_PROP_POS_FRAMES will always return `firstFrameIdx` value. I propose to remove the branch and forward the value from VideoCapture.

Reply:
> I have not found where iFrame is incremented besides the cese with initial seeking. CAP_PROP_POS_FRAMES will always return firstFrameIdx value.

@asmorkalov Sorry that was an oversite on my part, I've added the missing increment in the `VideoReaderImpl::internalGrab` and updated the test to check for this.

> I propose to remove the branch and forward the value from VideoCapture.

`cv::VideoCapture` is queing up frames for decoding in a different thread so its frame index will always be greater than or equal to the frame index realting to the frame returned from `cudacodec::VideoReader::nextFrame`.  This is explained in more detail including the reason for the branch in https://github.com/opencv/opencv_contrib/issues/3541.